packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -70,7 +70,89 @@ export class AndroidEmulatorDevice extends DeviceBase {
     }, DISPOSE_TIMEOUT);
   }
 
+  private async shouldUpdateLocale(newLocale: Locale) {
+    const locale = newLocale.replace("_", "-");
+    const { stdout } = await exec(ADB_PATH, [
+      "-s",
+      this.serial!,
+      "shell",
+      "settings",
+      "get",
+      "system",
+      "system_locales",
+    ]);
+
+    // if user did not use the device before it might not have system_locales property
+    // as en-US is the default locale we assume that no value is the same as en-US
+    if ((stdout ?? "en-US") === locale) {
+      return false;
+    }
+    return true;
+  }
+
+  // this method changes device locale in two ways, firstly it modifies system_locales setting,
+  // which in most cases is enough, to change locale. Unfortunately if persist.sys.locale exist
+  // it will conflict with system_locales and make device behavior unpredictable, so in that case
+  // we need to modify if as well. persist.sys.locale is added to the device when a user changes locale
+  // in the settings. Unfortunately this method does not handle all possible scenarios.
+  // If the device uses google mobile services persist.sys.locale can not be changed, as it requires a root access
+  // TODO:  Find a way to change or remove persist.sys.locale without root access. note: removing the whole
+  // data/property/persistent_properties file would also work as we persist device settings globally in Radon IDE
+  private async changeLocale(newLocale: Locale): Promise<boolean> {
+    const locale = newLocale.replace("_", "-");
+
+    await exec(ADB_PATH, [
+      "-s",
+      this.serial!,
+      "shell",
+      "settings",
+      "put",
+      "system",
+      "system_locales",
+      locale,
+    ]);
+
+    const { stdout } = await exec(ADB_PATH, [
+      "-s",
+      this.serial!,
+      "shell",
+      "getprop",
+      "persist.sys.locale",
+    ]);
+
+    if (!stdout) {
+      return true;
+    }
+
+    // If persist.sys.locale exist try to set it to match the newLocale.
+    try {
+      await exec(ADB_PATH, [
+        "-s",
+        this.serial!,
+        "shell",
+        "su",
+        "0",
+        "setprop",
+        "persist.sys.locale",
+        locale,
+      ]);
+    } catch (e) {
+      // this is expected if device is using google mobile services
+      Logger.debug(
+        "CHANGE LOCALE - Device is not allowing root access, moving on without modifying persist.sys.locale"
+      );
+    }
+
+    return true;
```

Comment:
I think it should be ok to remove this part. It is unlikely that someone would change the language via settings, and even if they do, it is unlikely you can actually use sudo on Android devices as it is only available on pure AOSP builds which are rarely used in practice